File: packages/tactical-engine/src/grid/coordinates.ts
import type { Coordinate } from "@limiarmap/shared-contracts";

export function coordinateKey({ x, y }: Coordinate): string {
  return `${x},${y}`;
}

export function isSameCoordinate(a: Coordinate, b: Coordinate): boolean {
  return a.x === b.x && a.y === b.y;
}

export function chebyshevDistance(a: Coordinate, b: Coordinate): number {
  return Math.max(Math.abs(a.x - b.x), Math.abs(a.y - b.y));
}

export function manhattanDistance(a: Coordinate, b: Coordinate): number {
  return Math.abs(a.x - b.x) + Math.abs(a.y - b.y);
}

File: packages/tactical-engine-py/tactical_engine/grid/coordinates.py
from shared_contracts import Coordinate


def coordinate_key(c: Coordinate) -> str:
    return f"{c.x},{c.y}"


def is_same_coordinate(a: Coordinate, b: Coordinate) -> bool:
    return a.x == b.x and a.y == b.y


def chebyshev_distance(a: Coordinate, b: Coordinate) -> int:
    return max(abs(a.x - b.x), abs(a.y - b.y))


def manhattan_distance(a: Coordinate, b: Coordinate) -> int:
    return abs(a.x - b.x) + abs(a.y - b.y)
